compiler/rustc_metadata/src/creader.rs

```diff
@@ -517,7 +517,7 @@ impl<'a> CrateLoader<'a> {
         self.maybe_resolve_crate(name, dep_kind, None).unwrap_or_else(|err| {
             let missing_core =
                 self.maybe_resolve_crate(sym::core, CrateDepKind::Explicit, None).is_err();
-            err.report(&self.sess, span, missing_core)
+            err.report_fatal(&self.sess, span, missing_core)
```

Comment:
It's preferable to return an `Option` from `resolve_crate`, so it can be used from `process_extern_crate` as well.
If you want an aborting version, you can create a new function `fn resolve_crate_or_abort` wrapping `resolve_crate` and `FatalError.raise()`-ing if it returns `None`.